Refactored main branch

nickolas-shtayn opened the pull request:
This was very difficult but I've learned a lot!

sstern6 reviewed (commented):
Awesome! 

So this is hard to review bc i dont know what you did or why you did it.

Can you make the pr description annoyingly detailed.

nickolas-shtayn commented:
@sstern6, 

Not sure how annoyingly detailed you want it, I can make it more detailed than this:


## HTML:

- Updated names for certain tags for clarity
- Replaced the backspace filled design with the original outline of backspace like the initial design

## CSS:
- Got rid of redundant `@import` statements
- Created variable for reusable properties (colors, calculator width & height, fonts, ratios for certain parts of the calculator)
- Got rid of redundant properties
- Made the calculator responsive using `clamp()` and using the width and height of the calculator as a base unit to be scaled off of

## JavaScript
### Abstractions:
- `handleCalculatorInput()` - handles user choice and assigns it to the right place
- `clearDisplay()` - handles any removal of calculation, whether clearing the screen or backspace
- `CalculateResult()` - performs the calculation and returns the result
- `updateDisplay()` - updates the display with the current calculation iteration
### Helper functions:
- `isNumeric()` - checks to see if an input is a numeric
- `isFirstNumber()` - checks to see if the current user choice is part of the first number or not

All of these work with both event types `"click"` or `"keydown"`.

nickolas-shtayn commented:
I will push a change to this branch for the `helpers.js` requirement and move those abstractions there with a `console.log()`

nickolas-shtayn commented:
I couldn't use `Node.js` syntax for exports so I had to use `ESModules` instead but I commented out what I would have done for the `Node.js` exports. I also got rid of the defer tag in the HTML because it was redundant with the `type="module"`
